### PR TITLE
feat: aggregation footprint

### PR DESCRIPTION
**Pull request by vitaligi**
implement a feature flag that performs smoothing of the aggregated footprint outline, as well as closing small holes (gaps)

| Question                | Answer                                                                          |
| ---------------- | -------------------------------------------------------------------------- |
| Bug fix         | ✖                                                                        |
| New feature     | ✔                                                                        |
| Breaking change | ✖                                                                        |
| Deprecations    | ✖                                                                        |
| Documentation   | ✖                                                                        |
| Tests added     | ✖                                                                        |
| Chore            | ✖                                                                       |

### CODE REVIEW

File: config/default.json [hunk 1]
```diff
@@ -41,7 +41,11 @@
       }
     },
     "aggregation": {
-      "maxDecimalDigits": 12
+      "maxDecimalDigits": 12,
+      "fixGeometry": {
+        "enabled": true,
+        "bufferSize": 0.000006
```
Comment (CL-SHLOMIKONCHA) [inline config/default.json:47]: bufferSizeDeg

File: helm/templates/configmap.yaml [hunk 2]
```diff
@@ -19,6 +19,10 @@ data:
   ENTITY_POLYGON_PARTS_NAME_PREFIX: {{ .Values.env.application.entities.polygonParts.namePrefix | quote }}
   ENTITY_POLYGON_PARTS_NAME_SUFFIX: {{ .Values.env.application.entities.polygonParts.nameSuffix | quote }}
   AGGREGATION_MAX_DECIMAL_DIGITS: {{ .Values.env.application.aggregation.maxDecimalDigits | quote }}
+  AGGREGATION_FIX_GEOMETRY_ENABLED: {{ .Values.env.application.aggregation.fixGeometry.enabled | quote }}
+  {{- if .Values.env.application.aggregation.fixGeometry.enabled }}
+  AGGREGATION_FIX_GEOMETRY_BUFFER_SIZE: {{ .Values.env.application.aggregation.fixGeometry.bufferSize | quote }}
```
Comment (CL-SHLOMIKONCHA) [inline helm/templates/configmap.yaml:24]: AGGREGATION_FIX_GEOMETRY_BUFFER_SIZE_DEG

File: src/aggregation/models/aggregationManager.ts [hunk 3]
```diff
@@ -76,23 +87,25 @@ export class AggregationManager {
...
-
-    const aggregationQuery = polygonPart.manager
-      .createQueryBuilder()
-      .addCommonTableExpression(aggregationCTE, 'aggregationCTE')
-      .select('*')
-      .addSelect(`trim(both '[]' from "aggregationCTE".footprint ->> 'bbox')`, 'productBoundingBox')
-      .from<AggregationLayerMetadata>('aggregationCTE', 'aggregationCTE');
+      }, 'sensors')
+      .addSelect(
+        `trim(both '[]' from (st_asgeojson(st_geometryn(st_collect("footprintUnionCTE".footprint_union), 1), maxdecimaldigits => ${this.maxDecimalDigits}, options => 1)::json ->> 'bbox'))`,
```
Comment (CL-SHLOMIKONCHA) [inline src/aggregation/models/aggregationManager.ts:104]: We need to make sure the result of bbox consider 'lon-lat' 'lat-lon'

**Review by CL-SHLOMIKONCHA — CHANGES_REQUESTED** [review]

File: config/default.json (hunk 1 above)
Comment (vitaligi) [reply config/default.json:47]: done

File: helm/templates/configmap.yaml (hunk 2 above)
Comment (vitaligi) [reply helm/templates/configmap.yaml:24]: done

File: src/aggregation/models/aggregationManager.ts (hunk 3 above)
Comment (vitaligi) [reply src/aggregation/models/aggregationManager.ts:104]: it's lon-lat (checked for 4326)

**Review by CL-SHLOMIKONCHA — APPROVED** [review]

**Pull request merged**